golang/pkg/dagent/utils/dockerhelper.go

```diff
@@ -5,34 +5,136 @@ package utils
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"strings"
 
 	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/events"
 	"github.com/docker/docker/client"
 	"github.com/rs/zerolog/log"
 
+	"github.com/dyrector-io/dyrectorio/golang/internal/grpc"
 	dockerHelper "github.com/dyrector-io/dyrectorio/golang/internal/helper/docker"
 	"github.com/dyrector-io/dyrectorio/golang/internal/label"
 	"github.com/dyrector-io/dyrectorio/golang/internal/mapper"
 	"github.com/dyrector-io/dyrectorio/golang/internal/util"
 	"github.com/dyrector-io/dyrectorio/protobuf/go/common"
 )
 
-func GetContainersByPrefix(ctx context.Context, prefix string) []*common.ContainerStateItem {
-	var containers []types.Container
-	var err error
+func getContainerIdentifierFromEvent(event *events.Message) *common.ContainerIdentifier {
+	prefix, hasValue := event.Actor.Attributes[label.DyrectorioOrg+label.ContainerPrefix]
+	if !hasValue {
+		prefix = ""
+	}
+
+	name, hasValue := event.Actor.Attributes["name"]
+	if !hasValue {
+		return nil
+	} else if len(prefix) > 0 {
+		name = strings.TrimPrefix(name, prefix+"-")
+	}
 
+	containerID := common.ContainerIdentifier{
+		Prefix: prefix,
+		Name:   name,
+	}
+
+	return &containerID
+}
+
+func eventToMessage(ctx context.Context, prefix string, event *events.Message) (*common.ContainerStateItem, error) {
+	if event.Type != "container" {
+		return nil, nil
+	}
+
+	containerID := getContainerIdentifierFromEvent(event)
+	if containerID == nil {
+		return nil, errors.New("event has no container name")
+	}
+
+	if prefix != "" && containerID.Prefix != prefix {
+		return nil, nil
+	}
+
+	if event.Action == "destroy" {
+		destroy := &common.ContainerStateItem{
+			Id:        containerID,
+			Command:   "",
+			CreatedAt: nil,
+			State:     common.ContainerState_REMOVED,
+			Reason:    "",
+			Status:    "",
+			Ports:     []*common.ContainerStateItemPort{},
+			ImageName: "",
+			ImageTag:  "",
+		}
+		return destroy, nil
+	}
+
+	containerState := mapper.MapDockerContainerEventToContainerState(event.Action)
+	if containerState == common.ContainerState_CONTAINER_STATE_UNSPECIFIED {
+		return nil, nil
```

Comment:
Let's write a comment about why we ignore some events, and provide some examples.